src/include/storage/metric/abstract_metric.h

```diff
@@ -0,0 +1,354 @@
+#pragma once
+
+#include <emmintrin.h>
+#include <atomic>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "catalog/catalog_defs.h"
+#include "storage/metric/abstract_raw_data.h"
+#include "storage/metric/metric_defs.h"
+
+namespace terrier {
+
+namespace transaction {
+class TransactionContext;
+}  // namespace transaction
```

Comment:
Is there a reason that you did this declaration here? I believe that this is unnecessary and it should compile without this.

Reply:
I ran `MetricTests` on my pc after removing this, and it compiled and the tests passed. I think you can remove `namespace transaction` and do `namespace terrier::storage::metric` instead.